### PR TITLE
#235 Reservation views create/delete

### PR DESCRIPTION
**Pull request by Vatavuk**
For #235, implemented create/delete view for reservations. Left two puzzles

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#290](https://github.com/timelesslounge/timelessis/issues/290) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#290](https://github.com/timelesslounge/timelessis/issues/290) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/5261839e-304a-4916-be40-9cbb9ae94627); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/reservations/views.py [hunk 1]
```diff
@@ -91,12 +96,15 @@ def edit(id):
...
 
-@bp.route("/delete", methods=["POST"])
-def delete():
-    flash("Delete not yet implemented")
+@bp.route("/delete/<int:id>", methods=["POST"])
+def delete(id):
+    """ Delete reservation """
+    reservation = Reservation.query.get(id)
+    DB.session.delete(reservation)
+    DB.session.commit()
```
Comment (amihaiemil) [inline timeless/reservations/views.py]: @Vatavuk We agreed at some point to not tie the DB stuff to the views. We have now ``CrudAPIView`` -- can you use that to abstract the DB away? As I remember, it was used already in Reservation Settings.

File: timeless/reservations/views.py [hunk 2]
```diff
@@ -76,13 +78,16 @@ def list_reservations(reservations):
...
-    if request.method == "POST":
-        flash("Create not yet implemented")
-
+def create():
+    """ Create new reservation """
+    form = ReservationForm(request.form)
+    form.validate()
+    if request.method == "POST" and form.validate():
+        form.save()
+        return redirect(url_for("reservations.list_reservations"))
```
Comment (amihaiemil) [inline timeless/reservations/views.py]: @Vatavuk Can we return ``BAD_REQUEST`` in an ``else``  here? That's how it was done in TableShape.edit() for instance. And it make more sense to tell the user that they did something wrong rather than simply listing the existing tables.

File: timeless/templates/reservations/create_edit.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+{% endblock %}
+
+{% block content %}
+  <form method="post">
+    {{ form }}
+    <p>
+      <button type="submit">Create</button>
+    </p>
+  </form>
+{% endblock %}
```
Comment (amihaiemil) [inline timeless/templates/reservations/create_edit.html]: @Vatavuk new line here. And why did you delete the ``create_edit.html`` file bellow?

File: timeless/reservations/views.py (hunk 1 above)
Comment (Vatavuk) [reply timeless/reservations/views.py]: @amihaiemil this is a template view not an API view. I suppose it will be refactored using GenericView and/or DB abstraction (#288) but these things are not yet implemented.

File: timeless/reservations/views.py (hunk 2 above)
Comment (Vatavuk) [reply timeless/reservations/views.py]: @amihaiemil I think this is a wrong approach, we should return a template containing errors to show the user validation problems directly on the browser. That's common practice (at least to my knowledge). We can render the errors in template using {{form.errors}}. I think I will raise an issue for this. If you don't agree we can ask `ARC` for opinion.

File: timeless/templates/reservations/create_edit.html (hunk 3 above)
Comment (Vatavuk) [reply timeless/templates/reservations/create_edit.html]: @amihaiemil I've moved it to the reservation directory

**Comment by Vatavuk** [comment]
@amihaiemil fixed and merged with master

**Comment by amihaiemil** [comment]
@Vatavuk Travis fails now, can you have a look?

File: timeless/reservations/views.py (hunk 2 above)
Comment (amihaiemil) [reply timeless/reservations/views.py]: @Vatavuk 
> I think I will raise an issue for this.

Makes sense, I think you can do that and discuss with the ARC there :)

**Comment by amihaiemil** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#290](https://github.com/timelesslounge/timelessis/issues/290) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@amihaiemil thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
